api/services/User/index.js

```diff
@@ -38,8 +38,8 @@ export default class User extends BaseModelService {
     }
   }
 
-  static async batchFunctionForAvailabilityLoader(keys) {
-    const resultPromise = keys.map((key) => {
+  static async findAllAvailabilityDuring(keys) {
+    const resultPromises = keys.map((key) => {
```

Comment:
this whole thing can be done in a single map over keys. Inside the map callback await for the db query and return !data.

```javascript
{
  return Promise.all(keys.map(async key => !(await DBQUERY())))
}
```